python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py

```diff
@@ -242,41 +241,31 @@ def on_message(client, userdata, msg):
     global _assetName
     try:
         if _callback_event_loop is None:
-            _LOGGER.debug("Message processing event doesn't yet exist - creating new event loop.")
             asyncio.set_event_loop(asyncio.new_event_loop())
             _callback_event_loop = asyncio.get_event_loop()
-        time_stamp = utils.local_timestamp()
-        try:
-            string_data = msg.payload.decode('utf-8')
-            dict_data = json.loads(string_data)
+        # Protobuf message structure
+        sparkplug_payload = sparkplug_b_pb2.Payload()
+        sparkplug_payload.ParseFromString(msg.payload)
+
+        for metric in sparkplug_payload.metrics:
+            value = ""
+            if metric.HasField("bool_value"):
+                value = metric.bool_value
+            elif metric.HasField("float_value"):
+                value = metric.float_value
+            elif metric.HasField("int_value"):
+                value = metric.int_value
+            elif metric.HasField("string_value"):
+                value = metric.string_value
+            # TODO: FOGL- 9198 - Handle other data types
             data = {
                 'asset': _assetName,
-                'timestamp': time_stamp,
-                'readings': dict_data if isinstance(dict_data, dict) else {"value": string_data}
+                'timestamp': datetime.fromtimestamp(metric.timestamp, tz=timezone.utc
+                                                    ).strftime('%Y-%m-%d %H:%M:%S.%s'),
+                'readings': {metric.name: value}
```

Comment:
For data types not supported, would we log some warning and ignore the metric ..?

Reply:
There were some problems with logging in plugin itself. Now it has been fixed and with unknown/not supported metric type readings are ignored.